## Setup

In [ ]:
import pandas as pd

## Programatically calculate the number of values in the "datetime" column of the UFO sightings dataset that can't be converted to actual datetimes:

In [ ]:
ufos_path = 'YOUR PATH HERE'
ufo_sightings = pd.read_csv(ufos_path)

<ipython-input-5-60ea0c9e0214>:2: DtypeWarning: Columns (5,9) have mixed types. Specify dtype option on import or set low_memory=False.
  ufo_sightings = pd.read_csv(ufos_path)


In [ ]:
datetimes_count = pd.to_datetime(ufo_sightings['datetime'],errors='coerce').count()
sightings_count = ufo_sightings.shape[0]
missing_datetime_values = sightings_count - datetimes_count
missing_datetime_values

694

**694 values in the "datetime" column of the UFO sightings dataset can't be converted to actual datetimes.**

## Use the `count` method to quality check the house sales dataset for missing values in any of the columns, or any values in the "date" column that can't be converted to a datetime data type:

In [ ]:
houses_path = 'YOUR PATH HERE'
house_sales = pd.read_csv(houses_path)

In [ ]:
house_sales.count()

id               21613
date             21613
price            21613
bedrooms         21613
bathrooms        21613
sqft_living      21613
sqft_lot         21613
floors           21613
waterfront       21613
view             21613
condition        21613
grade            21613
sqft_above       21613
sqft_basement    21613
yr_built         21613
yr_renovated     21613
zipcode          21613
lat              21613
long             21613
sqft_living15    21613
sqft_lot15       21613
dtype: int64

**There are no missing values in any of the columns of the house sales dataset.**

In [ ]:
datetimes_count = pd.to_datetime(house_sales['date'],errors='coerce').count()
house_sales_count = house_sales.shape[0]
missing_datetime_values = house_sales_count - datetimes_count
missing_datetime_values

0

**All values in the "date" column can be converted to a `datetime` data type.**